### PR TITLE
SAT:  skip backward compatibility tests on spec/catalog if previous and actual spec/catalog are identical

### PR DESCRIPTION
**Pull request by alafanechere**
## What
We don't need to run backward compatibility tests - that can be long-running - if the current and previous connector spec /catalog are identical.

Unblocking https://github.com/airbytehq/airbyte/pull/20237#discussion_r1047276082
## How
Skip spec backward compatibility tests if the current and previous connector spec are identical.

### CODE REVIEW

**Review by davydov-d — APPROVED** [review]
Thanks!

File: airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py [hunk 1]
```diff
@@ -83,7 +83,10 @@ class TestSpec(BaseTest):
     previous_spec_cache: ConnectorSpecification = None
 
     @pytest.fixture(name="skip_backward_compatibility_tests")
-    def skip_backward_compatibility_tests_fixture(self, inputs: SpecTestConfig, previous_connector_docker_runner: ConnectorRunner) -> bool:
+    def skip_backward_compatibility_tests_fixture(self, inputs: SpecTestConfig, previous_connector_docker_runner: ConnectorRunner,  previous_connector_spec: ConnectorSpecification, actual_connector_spec: ConnectorSpecification) -> bool:
+        if actual_connector_spec == previous_connector_spec:
+            pytest.skip("The previous and actual specifications are identical.")
```
[the remark this answers is not in the document]
Comment (alafanechere) [reply airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py:88]: We can do the same for backward compatibility test in `TestDiscover`: if the previous and actual discovered catalogs are identical we can skip the backward compatibility test.

[the remark this answers is not in the document]
Comment (erohmensing) [reply airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py:88]: Cool, just wanted to make sure that was the case 👍🏻

**Review by erohmensing — APPROVED** [review]

**Review by bazarnov — APPROVED** [review]
This is what there should be from the start! Thanks!

**Review by sherifnada — APPROVED** [review]

**Comment by alafanechere** [comment]
/publish connector=bases/source-acceptance-test auto-bump-version=false

**Pull request merged**